### ISSUE TITLE
Replace blacklist/whitelist throughout the PyTorch codebase

### ISSUE THREAD

**Issue by SplitInfinity** [pytorch/pytorch#41443]
## Goal
The goal of this issue is to track overall progress toward eliminating the use of the terms *whitelist* and *blacklist* from the PyTorch codebase. The recommended replacements for these terms are *allowlist* and *blocklist*, respectively. [See this numpy issue for more information.](https://github.com/numpy/numpy/pull/16669)

However, we should strive to use more specific and appropriate terms based on the context in which *whitelist* and *blacklist* are used. For example, if the code in question constructs a list of tests that are to be skipped and stores it into a variable called `test_blacklist`, `test_skiplist` would be a better choice than `test_blocklist`. But if there are no other options, *allowlist* and *blocklist* are the designated fallback terms.

## How To Help
1. Pick a file from the list below. Comment on the issue saying that you are working on it and assign it to yourself.
2. Work on your task. **Replace all instances of *blacklist* or *whitelist* as appropriate.** Submit a PR. **Make sure your PR complies with [the PyTorch code review values](https://github.com/pytorch/pytorch/wiki/Code-review-values)**.
3. Choose someone who is familiar with the code in question to review your PR. GitHub UI can help with this.
4. Merge your PR (or get someone from the team to merge it).

## Unexpected Issues
It is possible that a simple find-and-replace strategy will not work for some of the occurrences in the list below. A good example is `aten/src/ATen/gen.py`; this tool has command-line flags with `whitelist` in them that are consumed by other tools. We need to be more careful in these cases. For now, if you come across a case like this, leave a comment in the issue detailing your findings and edit this post to link the corresponding file to that issue or comment below.

## To-Do List

**Blacklist**
- [x] `.jenkins/pytorch/dirty.sh` (#41699)
- [x] `aten/src/ATen/native/cudnn/Conv.cpp` (#41700)
- [x] `caffe2/onnx/onnx_exporter.cc` (#41701)
- [x] `caffe2/onnx/onnx_exporter.h` (#41702)
- [x] `caffe2/opt/backend_transformer_base.h` (#41703)
- [x] `caffe2/opt/onnxifi_transformer.cc` (#41704)
- [x] `caffe2/opt/onnxifi_transformer.h` (#41705)
- [x] `caffe2/opt/tvm_transformer.cc` (#41706)
- [x] `caffe2/opt/tvm_transformer.h` (#41707)
- [ ] `caffe2/opt/custom/fakefp16_transform.cc` (#41708)
- [ ] `caffe2/opt/custom/glow_net_transform.cc` (#41709)
- [ ] `caffe2/opt/custom/glow_net_transform.h` (#41710)
- [x] `caffe2/python/pybind_state.cc` (#41711)
- [x] `caffe2/python/onnx/frontend.py` (#41712)
- [ ] `cmake/Modules_CUDA_fix/upstream/FindCUDA.cmake` (#41713)
- [ ] `cmake/public/cuda.cmake` (#41714)
- [x] `docs/cpp/source/Doxyfile` (#41715)
- [x] `test/run_test.py` (#41716)
- [x] `test/test_jit.py` (#41717)
- [x] `test/test_mobile_optimizer.py` (#41718)
- [x] `test/test_type_hints.py` (#41719)
- [x] `tools/autograd/gen_python_functions.py` (#41720)
- [x] `tools/jit/gen_unboxing_wrappers.py` (#41721)
- [x] `tools/pyi/gen_pyi.py` (#41722)
- [x] `torch/csrc/jit/passes/xnnpack_rewrite.cpp` (#41723)
- [x] `torch/csrc/jit/passes/xnnpack_rewrite.h` (#41724)
- [x] `torch/csrc/jit/python/init.cpp` (#41725)
- [x] `torch/csrc/utils/python_arg_parser.cpp` (#41726)
- [x] `torch/jit/_recursive.py` (#41727)
- [x] `torch/utils/mobile_optimizer.py` (#41728)
- [x] `torch/utils/hipify/hipify_python.py` (#41729)
- [x] `caffe2/onnx/onnx_exporter.h` (#41730)
- [x] `caffe2/opt/tvm_transformer.cc` (#41731)
- [ ] `caffe2/opt/custom/glow_net_transform.cc` (#41732)
- [x] `test/backward_compatibility/check_backward_compatibility.py` (#41733)
- [x] `caffe2/python/pybind_state.cc` (#41734)
- [x] `caffe2/python/onnx/onnxifi.py` (#41735)
- [ ] `torch/onnx/symbolic_helper.py` (#41736)
- [ ] `torch/onnx/symbolic_opset7.py` (#41737)
- [ ] `torch/onnx/symbolic_opset8.py` (#41738)

**Whitelist**
- [ ] `CMakeLists.txt` (#41739)
- [x] `.circleci/cimodel/data/simple/binary_smoketest.py` (#41740)
- [x] `.github/workflows/clang_format.yml` (#41741)
- [ ] `aten/src/ATen/gen.py` (#41742)
- [x] `caffe2/predictor/transforms.cc` (#41744)
- [x] `caffe2/transforms/common_subexpression_elimination.cc` (#41745)
- [x] `caffe2/transforms/common_subexpression_elimination.h` (#41746)
- [ ] `cmake/Codegen.cmake` (#41747)
- [x] `cmake/Whitelist.cmake` (#41748)
- [x] `test/test_namedtuple_return_api.py` (#41749)
- [x] `test/backward_compatibility/check_backward_compatibility.py` (#41750)
- [x] `test/quantization/test_quantized_op.py` (#41751)
- [x] `tools/clang_format_all.py` (#41752)
- [x] `tools/clang_format_ci.sh` (#41753)
- [x] `tools/code_analyzer/gen_op_registration_whitelist.py` (#41754)
- [x] `torch/jit/_script.py` (#41755)
- [x] `torch/quantization/default_mappings.py` (#41756)
- [x] `caffe2/CMakeLists.txt` (#41757)
- [x] `test/test_autograd.py` (#41758)
- [x] `torch/csrc/jit/passes/quantization/helper.cpp` (#41759)
- [x] `torch/csrc/jit/passes/lower_tuples.cpp` (#41761)
- [x] `torch/quantization/_numeric_suite.py` (#41762)
- [x] `torch/quantization/default_mappings.py` (#41763)
- [x] `torch/quantization/quantize.py` (#41764)

**Comment by morrme** [comment]
First: thank you. 

Next: the first file, ` .jenkins/pytorch/dirty.sh` , only contains the world `blacklist` in a comment. Do you want an issue created for that one, or may I just submit a PR with the change? In this case, the explanation phrase that accompanies the term may suffice as a replacement: 

> TODO: Replace this with a NEGATIVE regex that allows us to blacklist
 files (letting us skip builds when they are unnecessary)

**Comment by SplitInfinity** [comment]
@morrme 
Just submitting a PR is fine. I'm not sure if non-members can edit my post anyway so perhaps my procedure was poorly planned.

**Comment by morrme** [comment]
@SplitInfinity Do you still just want 1 file per PR?

**Comment by SplitInfinity** [comment]
Yes, that would be preferred so that each change is as small as possible. If a file uses a function defined elsewhere, then you'd have to make an exception.

**Comment by SplitInfinity** [comment]
Oops, not done yet.

**Comment by morrme** [comment]
@SplitInfinity Just curious: Do all of the PR's on this repo get "closed" and then merged "offline"?

**Comment by SplitInfinity** [comment]
Yes. We consider our internal PyTorch repository to be the source of truth, and PRs are imported to our internal repository using a tool called [shipit](https://github.com/facebook/fbshipit), merged, and then committed to this repository.

**Comment by morrme** [comment]
Thanks! Sorry if this is covered elsewhere!

**Comment by kosuruvarunkumar** [comment]
Hi, Is this still open for contributing?
If yes, can someone help me understand what the requirement is and how to submit a successful PR? if someone could help me with this, this could be my first step into open source.
Thank you.

**Comment by kosuruvarunkumar** [comment]
Hi, a question, sorry for repeating if this is already addressed.
As every pull request may not be linked to this issue id, there is a high chance of creating a PR with the same changes. There is a way to find whether a PR is raised for a change in a particular file or not.
Thanks

**Comment by muhdmud** [comment]
#41627

**Comment by muhdmud** [comment]
#41627 replace blacklist in aten/src/ATen/native/cudnn/Conv.cpp
Replaced blacklist to blocklist in the file specified.

**Comment by ari-hacks** [comment]
replaced blacklist with blocklist in test/test_type_hints.py #41644

**Comment by kosuruvarunkumar** [comment]
@SplitInfinity I raised a PR for changes in /torch/onxx/synbolic_helper.py
symbolic_helper.py is used in symbolic_opset7.py and symbolic_opset8.py
Initially thought of creating a separate PR for each file, but after submitting a PR, observed that checks failed because corresponding changes were not made in the other files. How to resolve the current PR's. Should I close the current PR's and create new one's or can I edit my PR's

**Comment by VinodSKumar** [comment]
There is no mention of contribution for the following files yet.
 caffe2/opt/backend_transformer_base.h
 caffe2/opt/onnxifi_transformer.cc
 caffe2/opt/onnxifi_transformer.h
 caffe2/opt/tvm_transformer.cc
 caffe2/opt/tvm_transformer.h  

If it is open for contribution, I could do the needful.

**Comment by SplitInfinity** [comment]
I made issues for each file. Please comment on the issue/assign it yourself before starting to work on it and your PRs to the issues.

**Comment by Br0wnHammer** [comment]
Replace Use of "Whitelist" in torch/quantization/default_mappings.py

**Comment by 0tist** [comment]
Hi, a quick question, as I am new to open-sourcing, for now, I can just pick any of the files that are labeled 'open' and submit a PR as per the guidelines mentioned above for the same?

**Comment by SplitInfinity** [comment]
Yes, but only submit a PR if there are no existing PRs. Any issue that doesn't have a check mark beside it is not done yet.

**Comment by namanshah01** [comment]
There are still a few prs that are open from more than a month ago, are these issues active? Can I work on them?

**Comment by SplitInfinity** [comment]
Hello everyone, the remaining cases are ones that are no longer suitable for open-source contributors because they require substantial internal changes. I am leaving the tasks open because they are not done but will remove the label that marks them as good for first time contributors.
